src/librustdoc/passes/html_tags.rs

```diff
@@ -0,0 +1,122 @@
+use super::{span_of_attrs, Pass};
+use crate::clean::*;
+use crate::core::DocContext;
+use crate::fold::DocFolder;
+use crate::html::markdown::opts;
+use rustc_hir::hir_id::HirId;
+use rustc_session::lint;
+use rustc_span::Span;
+use pulldown_cmark::{Event, Parser};
+
+pub const CHECK_UNCLOSED_HTML_TAGS: Pass = Pass {
+    name: "check-unclosed-html-tags",
+    run: check_unclosed_html_tags,
+    description: "detects unclosed HTML tags in doc comments",
+};
+
+struct UnclosedHtmlTagsLinter<'a, 'tcx> {
+    cx: &'a DocContext<'tcx>,
+}
+
+impl<'a, 'tcx> UnclosedHtmlTagsLinter<'a, 'tcx> {
+    fn new(cx: &'a DocContext<'tcx>) -> Self {
+        UnclosedHtmlTagsLinter { cx }
+    }
+}
+
+pub fn check_unclosed_html_tags(krate: Crate, cx: &DocContext<'_>) -> Crate {
+    let mut coll = UnclosedHtmlTagsLinter::new(cx);
+
+    coll.fold_crate(krate)
+}
+
+const ALLOWED_UNCLOSED: &[&str] = &[
+    "area", "base", "br", "col", "embed", "hr", "img", "input", "keygen", "link", "meta", "param",
+    "source", "track", "wbr",
+];
+
+fn drop_tag(cx: &DocContext<'_>, tags: &mut Vec<String>, tag_name: String, hir_id: HirId, sp: Span) {
+    // In case the tag wasn't in the queue, I guess it's not really an issue...
+    if let Some(pos) = tags.iter().position(|t| *t == tag_name) {
+        for _ in pos + 1..tags.len() {
+            if ALLOWED_UNCLOSED.iter().find(|&at| at == &tags[pos + 1]).is_some() {
```

Comment:
I don't understand what this is doing. `pos` is where the opening tag is , `pos+1..len()` is all following tags ... what is `pos+1`? I think this will give false negatives on
```html
<div>
  <hr>
```
Please add that as a test case.

Reply:
I use the vec as a queue, therefore, all items coming after `pos` are included into it. In this case, I go top to bottom to the included tags.

Reply:
Right, but why is `pos+1` special? You look at `ALLOWED_UNCLOSED` only for that tag.

Reply:
No, I look it at another place as well. ;)